src/components/PostDetails.tsx

```diff
@@ -1,121 +1,74 @@
 import React, { useEffect, useState } from 'react';
-import { Loader } from './Loader';
-import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  fetchCreatePostComment,
+  fetchDeletePostComment,
+  fetchPostComments,
+} from '../slices/commentSlice';
+import { NewCommentForm } from './NewCommentForm';
+import { Loader } from './Loader';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
+  const {
+    items: comments,
+    hasError,
+    loaded,
+  } = useAppSelector(state => state.comment);
+  const dispatch = useAppDispatch();
 
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+  useEffect(() => {
+    async function loadComments() {
+      setVisible(false);
+      await dispatch(fetchPostComments(post.id));
```

Comment:
move this function out from useEffect, and move the useEffect to the end of the code